### PR TITLE
Update packages and APIs

### PR DESCRIPTION
**Pull request by aleksandrlevochkin**

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -368,39 +372,260 @@ export const colorbrewer: IColorBrewer = <IColorBrewer>{
...
+                    value: TableHeatMap.BucketCountMaxLimit
+                },
+            } 
+        });
+
+        this.general.slices.push(bucketsSlice);
+    }
+
+    private initBucketsWithColorbrewer = (dataView: DataView): void => {
+        if (this.general.colorbrewer.value.toString() === "") {
```
[the remark this answers is not in the document]
Comment (MulyukovAidar) [reply src/settings.ts:576]: refactor this if condition

File: src/settings.ts [hunk 2]
```diff
@@ -368,39 +372,260 @@ export const colorbrewer: IColorBrewer = <IColorBrewer>{
...
+            this.general.colorbrewer.value = TableHeatMap.DefaultColorbrewer;
+        }
+
+        const colorbrewerArray: IColorArray = colorbrewer[this.general.colorbrewer.value];
+        
+        let minBucketNum: number = 0;
+        let maxBucketNum: number = 0;
+
+        for (let bucketIndex: number = TableHeatMap.BucketCountMinLimit; bucketIndex < TableHeatMap.ColorbrewerMaxBucketCount; bucketIndex++) {
+            if (minBucketNum === 0 && Object.prototype.hasOwnProperty.call(colorbrewerArray, bucketIndex.toString())) {
```
[the remark this answers is not in the document]
Comment (MulyukovAidar) [reply src/settings.ts:586]: share lint error log

[the remark this answers is not in the document]
Comment (MulyukovAidar) [reply src/settings.ts:586]: if (minBucketNum === 0 && colorbrewerArray[bucketIndex.toString()]){
                minBucketNum = bucketIndex;
            }
try this

[the remark this answers is not in the document]
Comment (aleksandrlevochkin) [reply src/settings.ts]: Fixed

File: src/settings.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleksandrlevochkin) [reply src/settings.ts]: Removed toString()

File: test/visualBuilder.ts [hunk 3]
```diff
@@ -38,11 +38,11 @@ export class TableHeatMapBuilder extends VisualBuilderBase<VisualClass> {
         return new VisualClass(options);
     }
 
-    public get mainElement(): JQuery {
-        return this.element.children("div.divTableHeatMap");
+    public get mainElement(): Element | null {
```
[the remark this answers is not in the document]
Comment (aleksandrlevochkin) [reply test/visualBuilder.ts]: Changed to HTMLElement

File: test/visualBuilder.ts [hunk 4]
```diff
@@ -38,7 +38,7 @@ export class TableHeatMapBuilder extends VisualBuilderBase<VisualClass> {
         return new VisualClass(options);
     }
 
-    public get mainElement(): Element | null {
+    public get mainElement(): HTMLElement | null {
         return this.element.querySelector("div.divTableHeatMap");
     }
```
Comment (MulyukovAidar) [inline test/visualBuilder.ts:44]: public get rects(): NodeListOf<Element> - use specific HTMLElement | SVGElement

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,3 @@
-## 3.1.1
```
Comment (MulyukovAidar) [inline CHANGELOG.md]: this should not be deleted, also add 3.1.2 patchnote

Comment (aleksandrlevochkin) [reply CHANGELOG.md]: Done

File: test/visualBuilder.ts (hunk 4 above)
Comment (aleksandrlevochkin) [reply test/visualBuilder.ts:44]: Fixed

**Review by MulyukovAidar — APPROVED** [review]

**Pull request merged**